WordPress/Classes/ViewRelated/Post/PrepublishingViewController.swift

```diff
@@ -72,9 +90,25 @@ class PrepublishingViewController: UITableViewController {
         setupPublishButton()
         setupFooterSeparator()
 
+        updatePublishButtonLabel()
         announcePublishButton()
 
         calculatePreferredContentSize()
+
+        configureKeyboardToggle()
+    }
+
+    /// Toggles `keyboardShown` as the keyboard notifications come in
+    private func configureKeyboardToggle() {
+        NotificationCenter.default.publisher(for: UIResponder.keyboardDidShowNotification)
+            .map { _ in return true }
+            .assign(to: \.keyboardShown, on: self)
+            .store(in: &cancellables)
```

Comment:
Noob user of `Combine` here so apologies if my question is dumb: what is the function of the `.store(in: &cancellables)` in this call?

Reply:
No problem! I've used RxSwift in the past, so I had some exposure to this a while back.

`assign` returns an `AnyCancellable` which is like the object you would get back when calling `NotificationCenter.addObserver(... using block:)` to hold a reference to the subscription. The lifecycle of this reference determines how long the subscription is around for. So deallocating it will remove the subscription (and cancel any operations), similar to a call `NotificationCenter.removeObserver` for an object + selector observer.

The `store` just adds the `AnyCancellable` to the set. The set (`cancellables`) is basically just a container which holds references to subscriptions. By adding it to a property on the object it will be deallocated at the same time as `PublishingViewController`.

Reply:
Oh, gotcha! Thank you for explaining it!